### PR TITLE
Added dynamic accessibility to environment for variables

### PR DESCRIPTION
**Pull request by calebkleveter**
```swift
let host = Environment.process.DATABASE_HOST
let port: Int? = Environment.process.DATABASE_PORT
```

### CODE REVIEW

File: Sources/ServiceKit/Environment/Environment.swift [hunk 1]
```diff
@@ -42,6 +43,13 @@ public struct Environment: Equatable {
         return ProcessInfo.processInfo.environment[key]
     }
 
+    /// Gets a key from the proccess environment
+    ///
+    ///     Environment.detect().DATABASE_USER
+    public subscript(dynamicMember member: String) -> String? {
+        return ProcessInfo.processInfo.environment[member]
```
Comment (Mordil) [inline Sources/ServiceKit/Environment/Environment.swift]: What's the overhead of accessing this? Could we try caching?

**Review by tanner0101 — CHANGES_REQUESTED** [review]
I think it would be better to put `@dynamicMemberLookup` on a sub-type, that way we don't risk conflicting with actual properties on `Environment`. I also think this should be accessible statically (no need for an _instance_ of environment).

Maybe something like:

```swift
import ServiceKit

let user = Environment.process.DATABASE_USER
print(user) // String?
```

Where `Environment.process` is a simple wrapper struct

File: Sources/ServiceKit/Environment/Environment.swift (hunk 1 above)
Comment (tanner0101) [reply Sources/ServiceKit/Environment/Environment.swift]: Process environment should only be accessed during boot-time, so I think performance doesn't really matter here. Caching would also increase memory usage which would be a downside even after boot.

Comment (Mordil) [reply Sources/ServiceKit/Environment/Environment.swift]: Comment withdrawn!

**Comment by calebkleveter** [comment]
@tanner0101 Mind taking another look at this?

File: Sources/ServiceKit/Environment/Proccess.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+extension Environment {
+    
+    /// The process information of an environment. Wraps `ProcessInto.processInfo`.
+    @dynamicMemberLookup public struct Process {
+        
+        /// The process information of the environment.
+        public let info: ProcessInfo
```
Comment (tanner0101) [inline Sources/ServiceKit/Environment/Proccess.swift]: We could keep this private (and maybe underscore it?). That would be useful incase anyone does `export info=foo`. 

```suggestion
        private let _info: ProcessInfo
```

File: Sources/ServiceKit/Environment/Proccess.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        
+        /// Creates a new `Process` wrapper for process information.
+        ///
+        /// - parameter info: The process info that the wrapper accesses. Defaults to `ProcessInto.processInfo`.
+        public init(info: ProcessInfo = .processInfo) {
```
Comment (tanner0101) [inline Sources/ServiceKit/Environment/Proccess.swift]: This init could be kept private as well. The only way people should be creating one of these is via `Environment.process`.

File: Tests/LinuxMain.swift [hunk 4]
```diff
@@ -3,4 +3,5 @@ import XCTest
 
 XCTMain([
     testCase(ServiceKitTests.allTests),
+    testCase(EnvironmentTests.allTests)
```
Comment (tanner0101) [inline Tests/LinuxMain.swift]: I'd prefer to keep this in ServiceKitTests as it's only one test case. Managing LinuxMain and allTests arrays is such a pain in the butt.

**Review by tanner0101 — CHANGES_REQUESTED** [review]

File: Sources/ServiceKit/Environment/Proccess.swift (hunk 3 above)
Comment (calebkleveter) [reply Sources/ServiceKit/Environment/Proccess.swift]: I had to make this internal so `Environment` could still initialize it.

**Review by tanner0101 — APPROVED** [review]
Great work on this @calebkleveter, thanks :)

**Comment by penny-coin** [comment]
Hey @calebkleveter, you just merged a pull request, have a coin! 

You now have 441 coins.

**Pull request merged**